### ISSUE TITLE
Gasstation problem with Biome decoration fix which disables already decorating patch 

### ISSUE THREAD

**Issue by quentin452** [FalsePattern/GasStation#5]
If i use spongemixin + mixingasm i don't get theses warns
biome decoration fix : https://www.curseforge.com/minecraft/mc-mods/biome-decoration-fix/

descriptions : 

when i use gasstation with biome decoration fix , i have warnings , the biome decoration fix patch is not applied and so i can still have the already decorating crash

mods list : 

00gasstation-mc1.7.10-0.3.5
alreadydecoratingfix-1.0a

crash(exemple) : 

[crash-2022-10-08_20.55.06-server.txt](https://github.com/FalsePattern/GasStation/files/9742439/crash-2022-10-08_20.55.06-server.txt)

warn(exemple) : 

```
[21:51:41] [Client thread/INFO] [STDOUT]: [net.tclproject.mysteriumlib.asm.core.MiscUtils$SystemLogHelper:warning:92]: [WARNING] Can not find the target method of fix: ASMFix: net.minecraft.world.biome.BiomeDecorator#func_150512_a(Lnet/minecraft/world/World;Ljava/util/Random;Lnet/minecraft/world/biome/BiomeGenBase;II) -> net.tclproject.mysteriumlib.asm.fixes.MysteriumPatchesFixesH#func_150512_a(Lnet/minecraft/world/biome/BiomeDecorator;Lnet/minecraft/world/World;Ljava/util/Random;Lnet/minecraft/world/biome/BiomeGenBase;II)Z, EnumReturnSetting=ON_TRUE, EnumReturnType=FIX_METHOD_RETURN_VALUE, InjectorFactory: net.tclproject.mysteriumlib.asm.core.FixInserterFactory$OnEnter, CreateMethod = false
```

logs : 

https://paste.ee/p/eybsQ

**Comment by quentin452** [comment]
temporary workaround fix, downgrade gasstation to 0.3.2,warns is introduced in 0.3.3

maybe caused by https://github.com/FalsePattern/GasStation/commit/ee767edf506ca6fe3b23f07575294d1c1e3bd582 commit or https://github.com/FalsePattern/GasStation/commit/629ca36e31681466824b2918b349b4defbcf7bdf ?

**Comment by DrParadox7** [comment]
It also conflicts with [Immersive Cavegen](https://modrinth.com/mod/icg):

```
[12:43:14] [Client thread/INFO] [STDOUT/Forge]: [net.tclproject.mysteriumlib.asm.core.MiscUtils$SystemLogHelper:debug:87]: [DEBUG] Injecting fixes into class net.minecraft.world.gen.feature.WorldGenSand.
[12:43:14] [Client thread/INFO] [STDOUT/Forge]: [net.tclproject.mysteriumlib.asm.core.MiscUtils$SystemLogHelper:debug:87]: [DEBUG] Fixed method net.minecraft.world.gen.feature.WorldGenSand#<init>(Lnet/minecraft/block/Block;I)
[12:43:14] [Client thread/INFO] [STDOUT/Forge]: [net.tclproject.mysteriumlib.asm.core.MiscUtils$SystemLogHelper:warning:92]: [WARNING] Can not find the target method of fix: ASMFix: net.minecraft.world.gen.feature.WorldGenSand#generate(Lnet/minecraft/world/World;Ljava/util/Random;III) -> net.tclproject.mysteriumlib.asm.fixes.MysteriumPatchesFixesSand#generate(Lnet/minecraft/world/gen/feature/WorldGenSand;Lnet/minecraft/world/World;Ljava/util/Random;III)Z, EnumReturnSetting=ALWAYS, EnumReturnType=FIX_METHOD_RETURN_VALUE, InjectorFactory: net.tclproject.mysteriumlib.asm.core.FixInserterFactory$OnEnter, CreateMethod = false

```

**Comment by DrParadox7** [comment]
> maybe caused by https://github.com/FalsePattern/GasStation/commit/ee767edf506ca6fe3b23f07575294d1c1e3bd582 commit or https://github.com/FalsePattern/GasStation/commit/629ca36e31681466824b2918b349b4defbcf7bdf ?

You can rule out https://github.com/FalsePattern/GasStation/commit/629ca36e31681466824b2918b349b4defbcf7bdf. 
I've built and tested the mod and it still occurs on its prior commit (tested with https://github.com/FalsePattern/GasStation/commit/6c55ef0060c4c72341789dff80d7ed3828fca69e)

**EDIT:** 
I've built and tested https://github.com/FalsePattern/GasStation/commit/de8e31f62c656627cc1da20d5f3c7222d24ac1d7 too but without issue:
```
[[Client thread/INFO] [STDOUT/Forge]: net.tclproject.mysteriumlib.asm.core.MiscUtils$SystemLogHelper:debug:87]: [DEBUG] Injecting fixes into class net.minecraft.world.gen.feature.WorldGenSand.
[Client thread/INFO] [STDOUT/Forge]: [net.tclproject.mysteriumlib.asm.core.MiscUtils$SystemLogHelper:debug:87]: [DEBUG] Fixed method net.minecraft.world.gen.feature.WorldGenSand#<init>(Lnet/minecraft/block/Block;I)
[Client thread/INFO] [STDOUT/Forge]: [net.tclproject.mysteriumlib.asm.core.MiscUtils$SystemLogHelper:debug:87]: [DEBUG] Fixed method net.minecraft.world.gen.feature.WorldGenSand#generate(Lnet/minecraft/world/World;Ljava/util/Random;III)
```
__I can confirm that https://github.com/FalsePattern/GasStation/commit/e84977bdebb3a78d2efb2025fc78b4c156b4a9bf is the culprit.__
Technically Mysteriumlib is an ASM so the title is still on point.

**Comment by quentin452** [comment]
warns fixed in 0.4.0
